### PR TITLE
feature: Add support for github action variables

### PR DESCRIPTION
**Pull request by wildsheepz**
Adding support for GitHub actions variables. eg. `vars.ExampleVar` in workflows.
#1558

### CODE REVIEW

**Comment by kuanyong-wong-partior** [comment]
@KnisterPeter, the `lint` check is failing because of complexity of a function:
image
However, this function is actually a very simple function
image

File: pkg/model/workflow.go [hunk 1]
```diff
@@ -155,6 +155,7 @@ type Job struct {
 	Uses           string                    `yaml:"uses"`
 	With           map[string]interface{}    `yaml:"with"`
 	RawSecrets     yaml.Node                 `yaml:"secrets"`
+	RawVars        yaml.Node                 `yaml:"vars"`
```
[the remark this answers is not in the document]
Comment (wildsheepz) [reply pkg/model/workflow.go:158]: Ah ok. I will remove the syntax extension for vars. I'll do some testing before updating

File: pkg/model/workflow.go [hunk 2]
```diff
@@ -235,6 +236,32 @@ func (j *Job) Secrets() map[string]string {
...
+
+	var val string
+	if !decodeNode(j.RawVars, &val) {
+		return false
+	}
+
+	return val == "inherit"
+}
+
+func (j *Job) Vars() map[string]string {
```
[the remark this answers is not in the document]
Comment (wildsheepz) [reply pkg/model/workflow.go]: removed this function

File: pkg/runner/expression.go [hunk 3]
```diff
@@ -483,3 +485,26 @@ func getWorkflowSecrets(ctx context.Context, rc *RunContext) map[string]string {
...
+
+		if vars == nil {
+			vars = map[string]string{}
+		}
+
+		for k, v := range vars {
+			vars[k] = rc.caller.runContext.ExprEval.Interpolate(ctx, v)
+		}
+
+		return vars
```
[the remark this answers is not in the document]
Comment (wildsheepz) [reply pkg/runner/expression.go]: updated this to inherit vars from caller and adding them to rc.Config.Vars

File: pkg/runner/expression_test.go [hunk 4]
```diff
@@ -229,6 +234,8 @@ func TestInterpolate(t *testing.T) {
 		{" ${{ env.KEY_WITH_UNDERSCORES }} ", " value_with_underscores "},
 		{"${{ secrets.CASE_INSENSITIVE_SECRET }}", "value"},
 		{"${{ secrets.case_insensitive_secret }}", "value"},
+		{"${{ vars.CASE_INSENSITIVE_VAR }}", "value"},
+		{"${{ vars.case_insensitive_var }}", "value"},
```
[the remark this answers is not in the document]
Comment (wildsheepz) [reply pkg/runner/expression_test.go:241]: Yes i missed this. Thanks for pointing it out.

File: pkg/model/workflow.go [hunk 5]
```diff
@@ -235,6 +236,32 @@ func (j *Job) Secrets() map[string]string {
 	return val
 }
 
+func (j *Job) InheritVars() bool {
```
[the remark this answers is not in the document]
Comment (wildsheepz) [reply pkg/model/workflow.go]: Removed this function

**Review by cplee — APPROVED** [review]

**Comment by askb** [comment]
@cplee This change is merged but only available on the master. Can we get a new release for this feature please?

**Comment by dwightgunning** [comment]
Version was bumped over the weekend however the release workflow [run](https://github.com/nektos/act/actions/runs/5428652209) failed.

**Pull request merged**